### ISSUE TITLE
vaadin-text-field's input has invalid aria-required attribute value

### ISSUE THREAD

**Issue by OlliTietavainenVaadin** [vaadin/vaadin-text-field#430]
### Description
In required `vaadin-text-field`s and derived fields, `aria-required` attribute should be `true` for required fields.

### Expected outcome
https://www.w3.org/TR/WCAG20-TECHS/ARIA2.html
> The aria-required property can have values of "true" or "false". For example, if a user must fill in an address field, then aria-required is set to "true".
> Example 1: ` <input type="text" name="usrname" id="usrname" aria-required="true"/>[*]`

### Actual outcome
`aria-required` is a valueless (boolean) attribute

### Demo
https://vaadin.com/components/vaadin-text-field/html-examples/text-field-validators-demos

**Comment by web-padawan** [comment]
Confirmed. Most likely a regression from adding slotted `input` support.

**Comment by Haprog** [comment]
This is also applicable to `aria-disabled`, `aria-readonly` and `aria-invalid` usage (in `vaadin-text-field-mixin.html`).

All of those 4 properties are currently being set in the same way as boolean attributes but afaik they should all be set to `"true"` string value when true (otherwise unset or `"false"`).

But I also came across this https://w3c.github.io/using-aria/#some-examples-of-redundant-aria which suggest that it's actually "a waste of time" to set `aria-required="true"` if you also have `required` set (at least in the context of `<input>`).

https://www.w3.org/TR/html-aria/#document-conformance-requirements-for-use-of-aria-attributes-in-html
See this table (here's a screenshot with some rows removed to show the column headers and highlight only the relevant rows):
![image](https://user-images.githubusercontent.com/75155/77168520-a43dcc80-6ac0-11ea-8e95-f914eccacf7d.png)
Though the text for `aria-disabled` here seems to be a bit contradictory (compared to the `aria-disabled`, `aria-readonly`). I opened an issue about it here for clarification: https://github.com/w3c/html-aria/issues/225

After reading the above info (and few other related articles I found) I would suggest to update `vaadin-text-field-mixin.html` so that it doesn't set  `aria-disabled`, `aria-readonly` and `aria-required` at all. Just keeping `aria-invalid` and making sure that one is set with the correct string `"true"` (when invalid) or unset/omitted (when valid).

I think these were probably added for historical reasons (older browsers and/or screen readers).

I tested running Lighthouse accessibility audit (from Chrome dev tools) and [axe-coconut](https://chrome.google.com/webstore/detail/axe-coconut-web-accessibi/iobddmbdndbbbfjopjdgadphaoihpojp) (extension) audit for a page that only had `<vaadin-text-field label="Required" required></vaadin-text-field>`. Both audits report the problem when `aria-required` is incorrectly set as a boolean attribute (with an implicit value of empty string) but if I update the implementation to not set `aria-required` at all, then they don't complain about it (when only `required` is set).

**Comment by Haprog** [comment]
This fix has been released in `vaadin-text-field` [v2.5.5](https://github.com/vaadin/vaadin-text-field/releases/tag/v2.5.5)
